src/main/java/org/folio/service/pieces/PieceStorageService.java

```diff
@@ -187,6 +179,15 @@ private Future<List<Piece>> getPieceChunkByLineIds(Collection<String> poLineIds,
       .map(PieceCollection::getPieces);
   }
 
+  static String getQueryForUserTenants(List<String> userTenants, String query) {
+    if (userTenants == null) {
+      return query;
+    }
+    userTenants.add(null);
+    var cql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
```

Comment:
If we have tenantId length for example 20 symbols, the calculation suggests that approximately 103 strings receivingTenantId=<<tenantId>>, each with a length of 20 characters, can be included in GET request without exceeding the 4KB URL limit.
For the largest consortia we have 61 tenant, so we should have some capacity for the original 'query'

Reply:
but to be on more safe side we can add checking setting into settings table(because not all consortia can use central ordering): ALLOW_ORDERING_WITH_AFFILIATED_LOCATIONS
Here is how this setting is set on UI: https://folio-dev-thunderjet-consortium.ci.folio.org/settings/consortia-settings/central-ordering
This also can speedup the sql query because this query would be more simple without checking receivingTenantId
This setting should use cache, it can not be disabled when it was enabled

Reply:
Thanks, now filters based on the setting